ballista/core/src/utils.rs

```diff
@@ -446,35 +446,13 @@ where
     D::Error: Into<StdError>,
 {
     let endpoint = tonic::transport::Endpoint::new(dst)?
-        .connect_timeout(Duration::from_secs(20))
-        .timeout(Duration::from_secs(20))
-        // Disable Nagle's Algorithm since we don't want packets to wait
-        .tcp_nodelay(true)
-        .tcp_keepalive(Option::Some(Duration::from_secs(3600)))
-        .http2_keep_alive_interval(Duration::from_secs(300))
-        .keep_alive_timeout(Duration::from_secs(20))
-        .keep_alive_while_idle(true);
-    endpoint.connect().await
-}
-
-pub async fn create_grpc_client_connection_configurable<D>(
-    dst: D,
-    connection_timeout: Duration,
-    timeout: Duration,
-    keep_alive_timeout: Duration,
-) -> std::result::Result<Channel, Error>
-where
-    D: std::convert::TryInto<tonic::transport::Endpoint>,
-    D::Error: Into<StdError>,
-{
-    let endpoint = tonic::transport::Endpoint::new(dst)?
-        .connect_timeout(connection_timeout)
-        .timeout(timeout)
+        .connect_timeout(Duration::from_secs(7))
```

Comment:
from 20 secs -> 7, p99 for replication latency usually < 3s, so we will be able to shrink p99 latency for fetch 3x without compromising query failure rate